### PR TITLE
UIDATIMP-373: Data import settings page's 4th pane for Match Profiles: Create MARC records' match-to section :: DONE

### PR DESCRIPTION
**Pull request by taras-tkachenko**
## Purpose:
- Implement `MARC field section` in  `Match Profiles` form `Existing Record` part (UIDATIMP-373).
- Implement `Static Value` section in `Match Profiles` form `Incoming Record` part for `TEXT` value type (UIDATIMP-352)
- Implement a mechanism to change sections on `Record Type` change (incoming and existing both).

## Approach:
- Implement *variable sections* set in the `<FlexibleForm>` rendering config and their rendering method in `<FlexibleForm:Control>` component.
- Make this mechanism `Form State` dependent (using particular state fields).

## Issues:
[UIDATIMP-373](https://issues.folio.org/browse/UIDATIMP-373)
[UIDATIMP-352](https://issues.folio.org/browse/UIDATIMP-352)

### CODE REVIEW

File: src/components/FlexibleForm/Control.js [hunk 1]
```diff
@@ -216,6 +236,41 @@ export const Control = memo(props => {
...
+    const {
+      sectionNamespace: sectionNS,
+      stateFieldValue,
+      acceptedSections,
+      repeatableIndex: ri,
+    } = attribs;
+
+    const currentSection = commonSections.find(item => item.sectionKey === acceptedSections[stateFieldValue]);
+
+    if (!currentSection || !currentSection.controlType) {
```
Comment (VictorSoroka1) [inline src/components/FlexibleForm/Control.js]:
```suggestion
    if (!currentSection?.controlType) {
```

File: src/components/FlexibleForm/Control.js [hunk 2]
```diff
@@ -216,6 +236,41 @@ export const Control = memo(props => {
...
+        referenceTables={referenceTables}
+        componentsProps={componentsProps}
+        commonSections={commonSections}
+        record={record}
+        {...currentSection}
+      />
+    );
+  };
+
+  if (controlType === 'CommonSection') {
```
[the remark this answers is not in the document]
Comment (taras-tkachenko) [reply src/components/FlexibleForm/Control.js]: CommonSection is not a real component? it is an abstract container to render common controls.
I don't understand how and why do I need to move this to variable (which one?).

[the remark this answers is not in the document]
Comment (VictorSoroka1) [reply src/components/FlexibleForm/Control.js]: I just meant that 'CommonSection' string is hardcoded. The following way I see as an improvements:
- store `CommonSection` to a constant;
- store `controlType === 'CommonSection'` check to a variable like `isCommoneSectionType`;

File: src/components/FlexibleForm/Control.js [hunk 3]
```diff
@@ -61,10 +66,20 @@ export const Control = memo(props => {
   const isFragment = (isEditable && controlType === 'Fragment') || (!isEditable && staticControlType === 'Fragment');
 
   const getOptions = options => {
-    return options.map(option => ({
-      value: option.value,
-      label: intl.formatMessage({ id: option.label }),
-    }));
+    return options.map(option => {
+      const lbl = option.label;
```
Comment (VictorSoroka1) [inline src/components/FlexibleForm/Control.js]:
```suggestion
      const { label } = option;
```
In my opinion, the variables like `lbl` or `nsName` can be unreadable at some point. Now, it does not pose a problem as the methods are small. Overall, I would prefer readability over brevity.

**Pull request merged**